jwst/background/background_step.py

```diff
@@ -58,13 +57,37 @@ def process(self, input, bkg_list):
                 result = background_sub.subtract_wfss_bkg(
                                 input_model, bkg_name, wlrange_name)
                 result.meta.cal_step.back_sub = 'COMPLETE'
-
             else:
+                # check if input data is NRS_IFU
+                tolerance = 1.0e-8
+                do_sub = True
+                if input_model.meta.instrument.name in ["NIRSPEC"]:
+                    # check if GWA_XTIL & GWA_YTIL values of source
+                    # background are the same. If not skip step
+                    input_xtilt = input_model.meta.instrument.gwa_xtilt
+                    input_ytilt = input_model.meta.instrument.gwa_ytilt
+                    for bkg_file in bkg_list:
+                        bkg_model = datamodels.open(bkg_file)
```

Comment:
it would be better to use the the `with` context manager here so that you don't have to call `close()` a few lines down.
```
with datamodels.open(bkg_file) as bkg_model:
    bkg_xtilt = bkg_model.meta.instrument.gwa_xtilt
    bkg_ytilt = bkg_model.meta.instrument.gwa_ytilt
```